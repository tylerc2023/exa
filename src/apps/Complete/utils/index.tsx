export * from './configs';
export * from './useKeyboardHeight';
export * from './userFactory';
