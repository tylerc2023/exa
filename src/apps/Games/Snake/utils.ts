export type BoardObject = Array<Array<0 | 1 | 2>>;

export const generateRandom = (size: number) =>
  Math.floor(Math.random() * size);

export const addFood = (
  board: BoardObject,
  used: {[key: string]: boolean} = {},
): boolean => {
  const x = generateRandom(board.length);
  const y = generateRandom(board.length);

  const combo = `${x}${y}`;
  if (Object.keys(used).length === board.length) {
    return false;
  }
  if (combo in used || board[x][y] !== 0) {
    used[combo] = true;
    addFood(board, used);
    return false;
  }
  board[y][x] = 2;
  return true;
};

export const addStarting = (board: BoardObject) => {
  const center = Math.floor(board.length / 2);
  board[center][center] = 1;
};

export const generateBoard = (size: number) => {
  const board: BoardObject = [];
  for (let i = 0; i < size; i++) {
    board[i] = [];
    for (let j = 0; j < size; j++) {
      board[i][j] = 0;
    }
  }

  addStarting(board);
  addFood(board);
  return board;
};

export const nextSnakePosition = () => undefined;
export const collision = (next: any) => next && false;
export const endGame = () => undefined;
export const eat = () => undefined;
