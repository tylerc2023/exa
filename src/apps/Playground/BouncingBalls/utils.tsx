import {Animated} from 'react-native';
import {State} from './BouncingBall';
import {CanvasDimensions} from './Canvas';

const getRandomNumber = (min: number, max: number): number =>
  Math.random() * (max - min) + min;

type GetSpeedProps = {
  state: State;
  canvas: CanvasDimensions;
  diameter: number;
  pulse: boolean;
};

export const getSpeed = ({state, canvas, diameter, pulse}: GetSpeedProps) => {
  const dMax = state.diameter >= diameter * 1.1;
  const dMin = state.diameter <= diameter * 0.9;
  // TODO: recenter so pulsing happens from middle
  const growing = dMax || dMin ? !state.growing : state.growing;
  const d = state.diameter + (!pulse ? 0 : growing ? 1 : -1);
  const xMin = canvas.x;
  const xMax = canvas.x + canvas.width;
  const yMin = canvas.y;
  const yMax = canvas.y + canvas.height;
  const xLarger = state.x + d >= xMax;
  const xSmaller = state.x <= xMin;
  const yLarger = state.y + d >= yMax;
  const ySmaller = state.y <= yMin;
  // TODO: add collision
  const xSpeed = xLarger || xSmaller ? -state.xSpeed : state.xSpeed;
  const ySpeed = yLarger || ySmaller ? -state.ySpeed : state.ySpeed;
  const x = xLarger ? xMax - d : xSmaller ? xMin : state.x + xSpeed;
  const y = yLarger ? yMax - d : ySmaller ? yMin : state.y + ySpeed;
  const dx = x + xSpeed;
  const dy = y + ySpeed;
  return {xSpeed, ySpeed, x: dx, y: dy, d, growing};
};

type GetInitialStateProps = {
  canvas: CanvasDimensions;
  diameter: number;
  speed: number;
};

export const getInitialState = ({
  canvas,
  diameter,
  speed,
}: GetInitialStateProps): State => {
  const starting = {
    x: getRandomNumber(canvas.x, canvas.x + canvas.width - diameter),
    y: getRandomNumber(canvas.y, canvas.y + canvas.height - diameter),
  };
  const growing = !!Math.floor(Math.random());
  return {
    ...starting,
    position: new Animated.ValueXY(starting),
    xSpeed: getRandomNumber(-speed, speed),
    ySpeed: getRandomNumber(-speed, speed),
    diameter,
    growing,
  };
};
