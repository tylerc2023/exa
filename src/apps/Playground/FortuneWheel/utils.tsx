type GetRandomRange = {
  min: number;
  max: number;
};

const getRandomRange = ({min, max}: GetRandomRange) => {
  return Math.floor(Math.random() * (max - min) + min);
};

type GetNewLocation = {
  minSpin: number;
  maxSpin: number;
  numOfSegments: number;
  location: number;
};

export const getNewLocation = ({
  minSpin,
  maxSpin,
  numOfSegments,
  location,
}: GetNewLocation) => {
  const spins = getRandomRange({min: minSpin, max: maxSpin}) * 360;
  const winner = Math.floor(Math.random() * numOfSegments);
  const winnerSection = Math.floor(winner * (360 / numOfSegments));
  const newLocation = location + spins + winnerSection;
  return newLocation;
};

type GetWinnerIndex = {
  location: number;
  numOfSegments: number;
  angleOfSegment: number;
};

export const getWinnerIndex = ({
  location,
  numOfSegments,
  angleOfSegment,
}: GetWinnerIndex) => {
  const deg = Math.abs(Math.round(location % 360));
  const index =
    (numOfSegments - Math.floor(deg / angleOfSegment)) % numOfSegments;
  return index;
};
