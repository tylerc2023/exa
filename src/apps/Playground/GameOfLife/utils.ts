const operations = [
  [0, 1],
  [0, -1],
  [1, -1],
  [-1, 1],
  [1, 1],
  [-1, -1],
  [1, 0],
  [-1, 0],
];

type BoardItem = 0 | 1;

export type Board = Array<Array<BoardItem>>;

export const generateBoard = (size: number, random = 0): Board => {
  const board: Board = [];
  for (let i = 0; i < size; i++) {
    board[i] = [];
    for (let j = 0; j < size; j++) board[i][j] = Math.random() > random ? 0 : 1;
  }
  return board;
};

export const determineBoardItem = (
  board: Board,
  i: number,
  j: number,
): BoardItem => {
  let neighbors = 0;
  const item = board[i][j];
  const length = board.length;
  for (let k = 0; k < operations.length; k++) {
    const [x, y] = operations[k];
    const newI = i + x;
    const newK = j + y;
    if (newI >= 0 && newI < length && newK >= 0 && newK < length)
      neighbors += board[newI][newK];
    if (neighbors > 3) break;
  }

  return neighbors < 2 || neighbors > 3
    ? 0
    : item === 0 && neighbors === 3
    ? 1
    : item;
};
