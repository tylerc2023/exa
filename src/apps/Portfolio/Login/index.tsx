import {useNavigation} from '@react-navigation/native';
import React, {memo, useCallback, useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {Button, Screen, TextInput} from '../../../components';

const styles = StyleSheet.create({
  row: {flexDirection: 'row', justifyContent: 'space-around'},
});

export const Login = memo(function PortfolioLogin() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const {goBack, navigate} = useNavigation();
  const handlePassword = useCallback((val: string) => setPassword(val), []);
  const handleEmail = useCallback((val: string) => setEmail(val), []);
  const navBack = useCallback(() => goBack(), [goBack]);
  const navPortfolio = useCallback(() => navigate('home'), [navigate]);
  const navPassword = useCallback(() => navigate('forgotPassword'), [navigate]);

  return (
    <Screen gutter onLeftPress={navBack} title="Login">
      <TextInput
        onChangeText={handleEmail}
        placeholder="example@gmail.com"
        title="email"
        value={email}
      />
      <TextInput
        onChangeText={handlePassword}
        placeholder="•••••••"
        title="password"
        value={password}
      />
      <View style={styles.row}>
        <Button onPress={navPortfolio} title="Login" />
        <Button onPress={navPassword} title="Forgot" />
      </View>
    </Screen>
  );
});
