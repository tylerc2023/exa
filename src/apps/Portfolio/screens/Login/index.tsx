import React, { memo, useState } from "react";
import { Button, Screen, TextInput } from "../../../../components";
import { StyleSheet, View } from "react-native";
import { useNav } from "../../../../hooks";

const styles = StyleSheet.create({
  row: { flexDirection: "row", justifyContent: "space-around" }
});

export default memo(function PortfolioLogin() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const nav = useNav();
  const handlePassword = (val: string) => setPassword(val);
  const handleEmail = (val: string) => setEmail(val);

  return (
    <Screen onLeftPress={nav.to("portfolioLanding")} title="Login" gutter>
      <TextInput
        title="email"
        placeholder="example@gmail.com"
        onChangeText={handleEmail}
        value={email}
      />
      <TextInput
        title="password"
        placeholder="•••••••"
        onChangeText={handlePassword}
        value={password}
      />
      <View style={styles.row}>
        <Button title="Login" onPress={nav.to("portfolio")} />
        <Button title="Forgot" onPress={nav.to("portfolioForgotPassword")} />
      </View>
    </Screen>
  );
});
