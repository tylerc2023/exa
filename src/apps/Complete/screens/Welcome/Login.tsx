import {FirebaseAuthTypes} from '@react-native-firebase/auth';
import React, {memo, useCallback, useEffect, useRef, useState} from 'react';
import {Platform, TextInput as OriginalTextInput} from 'react-native';
import {
  ActivityIndicator,
  Button,
  Icon,
  Modal,
  Text,
  TextInput,
  TouchableOpacity,
} from '../../../../components';
import {useAuth, useColor} from '../../../../hooks';
import {config} from '../../../../utils';
import {ModalHeader} from '../../components/';

interface Props {
  onBackgroundPress: () => void;
  showFacebook?: boolean;
  showGoogle?: boolean;
  showApple?: boolean;
  showEmail?: boolean;
  showPhone?: boolean;
  showAnonymous?: boolean;
  onSuccess: (user: FirebaseAuthTypes.User | null) => void;
}

// TODO: relocate to somewhere
interface SignInButtonProps {
  onPress: () => void;
  icon: string;
  title: string;
  margin?: boolean;
}

const SignInButton = memo(function SignInButton({
  onPress,
  icon,
  title,
  margin,
}: SignInButtonProps) {
  const color = useColor();
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'row',
        borderColor: color.secondary,
        borderWidth: 1,
        padding: config.padding(2),
        marginBottom: config.padding(margin ? 4 : 0),
        borderRadius: config.sizing.borderRadius,
      }}>
      <Icon name={icon} />
      <Text
        style={{paddingLeft: config.padding(2)}}
        title={title}
        type="button"
      />
    </TouchableOpacity>
  );
});

type Screen =
  | 'loading'
  | 'landing'
  | 'signUp'
  | 'logIn'
  | 'forgotPassword'
  | 'phone'
  | 'phoneCode'
  | 'anonymous';

type Ref = {
  email: string;
  password: string;
  phone: string;
  phoneCode: string;
};

const initialRef: Ref = {
  email: '',
  password: '',
  phone: '',
  phoneCode: '',
};

type State = {
  eye: boolean;
  completeForm: boolean;
  passwordError: boolean;
  screen: Screen;
};

const initialState: State = {
  eye: false,
  completeForm: false,
  passwordError: false,
  screen: 'landing',
};

export const Login = memo(function Login({
  onBackgroundPress,
  showFacebook,
  showGoogle,
  showApple,
  showPhone,
  showEmail,
  showAnonymous,
  onSuccess,
}: Props) {
  const {
    initializing,
    user,
    error,
    loading,
    onErrorClear,
    onApple,
    onGoogle,
    onFacebook,
    onPhoneConfirm,
    onPhone,
    onEmail,
    onPasswordReset,
    onAnonymous,
  } = useAuth();
  const color = useColor();
  const successful = useRef(false);
  const form = useRef<Ref>(initialRef);
  const [state, setState] = useState<State>(initialState);
  const emailRef = useRef<OriginalTextInput | null>(null);
  const passwordRef = useRef<OriginalTextInput | null>(null);
  const eyeIcon = state.eye ? 'eye-outline' : 'eye-off-outline';

  const onFormChange = useCallback(
    (key: keyof Ref) => (val: string) => {
      form.current = {...form.current, [key]: val};
    },
    [],
  );

  const onPhonePress = useCallback(() => {
    onPhone(form.current.phone);
  }, [onPhone]);

  const onEye = useCallback(() => {
    setState((p) => ({...p, eye: !p.eye}));
    passwordRef.current?.focus();
  }, []);

  const onSubmitEditing = useCallback(
    (key: keyof typeof initialRef) => () => {
      if (key === 'email') passwordRef.current?.focus();
      if (key === 'password')
        onEmail(form.current.email, form.current.password);
    },
    [onEmail],
  );

  const onScreenChange = useCallback(
    (screen: Screen) => () => setState((p) => ({...p, screen})),
    [],
  );

  useEffect(() => {
    if (user && !successful.current) {
      onSuccess(user);
      successful.current = true;
    }
  }, [onSuccess, user]);

  return (
    <Modal onBackgroundPress={onBackgroundPress} showOverlay>
      <>
        {error && <Text center color="danger" title={error} />}
        {state.screen === 'loading' || initializing ? (
          <ActivityIndicator />
        ) : state.screen === 'phoneCode' ? (
          <>
            <TextInput
              keyboardType="number-pad"
              onChangeText={onFormChange('phoneCode')}
              placeholder="phone confirmation code"
            />
            <Button
              color="primary"
              emphasis="high"
              onPress={onPhoneConfirm(form.current.phoneCode)}
              title="verify phone confirmation code"
            />
            <Button onPress={onScreenChange('landing')} title="go back" />
          </>
        ) : state.screen === 'phone' ? (
          <>
            <ModalHeader
              onRightPress={onScreenChange('landing')}
              title="Phone Confirmation"
            />
            <TextInput
              autoCorrect={false}
              blurOnSubmit={false}
              keyboardType="number-pad"
              onChangeText={onFormChange('phone')}
              onRef={emailRef}
              onSubmitEditing={onSubmitEditing('phone')}
              placeholder="Phone number"
              returnKeyType="send"
              style={{marginBottom: config.padding(4)}}
              value=""
            />
            <Button
              center
              color="primary"
              emphasis="high"
              onPress={onPhonePress}
              title="send confirmation code"
            />
          </>
        ) : state.screen === 'forgotPassword' ? (
          <>
            <ModalHeader
              onRightPress={onScreenChange('logIn')}
              title="Reset password"
            />
            <Text
              center
              style={{paddingBottom: config.padding(4)}}
              title="Please enter the email address associated with your account."
            />
            <Text
              center
              style={{paddingBottom: config.padding(4)}}
              title="We'll send you an email with your login email as well as a password reset instructions."
            />
            <TextInput
              autoCorrect={false}
              blurOnSubmit={false}
              keyboardType="email-address"
              onChangeText={onFormChange('email')}
              onRef={emailRef}
              onSubmitEditing={onSubmitEditing('email')}
              placeholder="Email address"
              returnKeyType="send"
              style={{marginBottom: config.padding(4)}}
              textContentType="username"
              value=""
            />
            <Button
              center
              color="primary"
              emphasis="high"
              onPress={onPasswordReset(form.current.email)}
              title="Reset password"
            />
          </>
        ) : state.screen === 'signUp' ? (
          <>
            <ModalHeader
              onRightPress={onScreenChange('landing')}
              title="Sign up"
            />
            <TextInput
              autoCorrect={false}
              blurOnSubmit={false}
              editable={!loading}
              keyboardType="email-address"
              onChangeText={onFormChange('email')}
              onRef={emailRef}
              onSubmitEditing={onSubmitEditing('email')}
              placeholder="Email address"
              returnKeyType="next"
              style={{marginBottom: config.padding(4)}}
              textContentType="username"
              value=""
            />
            <TextInput
              autoCorrect={false}
              blurOnSubmit={false}
              editable={!loading}
              icons={[{name: eyeIcon, onPress: onEye, focus: true}]}
              onChangeText={onFormChange('password')}
              onRef={passwordRef}
              onSubmitEditing={onSubmitEditing('password')}
              placeholder="Password"
              returnKeyType="done"
              secureTextEntry={!state.eye}
              style={{marginBottom: config.padding(4)}}
              textContentType="password"
              value=""
            />
            <Button
              buttonStyle={{marginBottom: config.padding(4)}}
              center
              color="primary"
              disable={loading}
              lowercase
              onPress={onScreenChange('logIn')}
              title="Already have an account?"
            />
            <Button
              center
              color="primary"
              disable={loading}
              emphasis="high"
              onPress={onSubmitEditing('password')}
              title="Sign up"
            />
          </>
        ) : state.screen === 'logIn' ? (
          <>
            <ModalHeader
              onRightPress={onScreenChange('signUp')}
              title="Log in"
            />
            <TextInput
              backgroundColor={color.surface}
              keyboardType="email-address"
              onChangeText={onFormChange('email')}
              placeholder="email"
              style={{marginBottom: config.padding(4)}}
              textContentType="username"
            />
            <TextInput
              backgroundColor={color.surface}
              onChangeText={onFormChange('password')}
              placeholder="password"
              secureTextEntry
              style={{marginBottom: config.padding(4)}}
              textContentType="password"
            />
            <Button
              buttonStyle={{marginBottom: config.padding(4)}}
              center
              color="primary"
              lowercase
              onPress={onScreenChange('forgotPassword')}
              title="Forgot password?"
            />
            <Button
              center
              color="primary"
              emphasis="high"
              onPress={onEmail(form.current.email, form.current.password)}
              title="Log in"
            />
          </>
        ) : (
          <>
            <ModalHeader onRightPress={onBackgroundPress} title="Get Started" />
            {(Platform.OS === 'ios' || showApple) && (
              <SignInButton
                icon="apple"
                margin
                onPress={onApple}
                title="Continue with Apple"
              />
            )}
            {showGoogle && (
              <SignInButton
                icon="google"
                margin
                onPress={onGoogle}
                title="Continue with Google"
              />
            )}
            {showFacebook && (
              <SignInButton
                icon="facebook"
                margin
                onPress={onFacebook}
                title="Continue with Facebook"
              />
            )}
            {showEmail && (
              <SignInButton
                icon="email"
                margin
                onPress={onScreenChange('signUp')}
                title="Continue with Email"
              />
            )}
            {showPhone && (
              <SignInButton
                icon="phone"
                margin
                onPress={onScreenChange('phone')}
                title="Continue with Phone"
              />
            )}
            {showAnonymous && (
              <SignInButton
                icon="account"
                onPress={onAnonymous}
                title="Continue Anonymously"
              />
            )}
          </>
        )}
      </>
    </Modal>
  );
});
