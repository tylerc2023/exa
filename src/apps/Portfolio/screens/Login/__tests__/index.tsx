import * as React from "react";
import { Login } from "../../Login";
import { mockRenderer } from "../../../../../utils/Mock";

it("renders correctly", () => {
  const dom = mockRenderer(<Login />).toJSON();
  expect(dom).toMatchSnapshot();
});
