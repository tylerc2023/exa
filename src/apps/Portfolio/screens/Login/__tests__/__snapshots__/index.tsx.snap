// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 100%)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "hsl(208, 7%, 46%)",
        "borderBottomWidth": 0,
        "flexDirection": "row",
        "height": 72,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.2,
          "padding": 12,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "hsl(208, 7%, 46%)",
                "fontSize": 36,
              },
              Array [
                undefined,
                undefined,
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "hsla(0, 0%, 0%, 1)",
          "fontSize": 40,
          "fontWeight": "100",
          "letterSpacing": -0.5,
          "opacity": 1,
        }
      }
    >
      Login
    </Text>
    <View
      style={
        Object {
          "flex": 0.2,
          "padding": 12,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "hsl(0, 0%, 100%)",
          "flex": 1,
        },
        Object {
          "padding": 16,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 0,
              "paddingHorizontal": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(0, 0%, 0%)",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0.25,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            email
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          disableFullscreenUI={true}
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="example@gmail.com"
          placeholderTextColor="hsl(208, 7%, 46%)"
          rejectResponderTermination={true}
          selectionColor="hsl(211, 100%, 50%)"
          style={
            Array [
              Object {
                "backgroundColor": "hsl(0, 0%, 100%)",
                "borderColor": "hsl(208, 7%, 46%)",
                "borderRadius": 4,
                "borderWidth": 2,
                "color": "hsl(0, 0%, 0%)",
                "padding": 8,
                "paddingRight": 32,
                "width": "100%",
              },
              undefined,
              undefined,
              Object {
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0.25,
              },
              undefined,
            ]
          }
          textContentType="none"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "hsl(354, 70%, 54%)",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "opacity": 0,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 0,
              "padding": 0,
              "paddingHorizontal": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(354, 70%, 54%)",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0.25,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 0,
              "paddingHorizontal": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(0, 0%, 0%)",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0.25,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            password
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          disableFullscreenUI={true}
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="•••••••"
          placeholderTextColor="hsl(208, 7%, 46%)"
          rejectResponderTermination={true}
          selectionColor="hsl(211, 100%, 50%)"
          style={
            Array [
              Object {
                "backgroundColor": "hsl(0, 0%, 100%)",
                "borderColor": "hsl(208, 7%, 46%)",
                "borderRadius": 4,
                "borderWidth": 2,
                "color": "hsl(0, 0%, 0%)",
                "padding": 8,
                "paddingRight": 32,
                "width": "100%",
              },
              undefined,
              undefined,
              Object {
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0.25,
              },
              undefined,
            ]
          }
          textContentType="none"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "hsl(354, 70%, 54%)",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "opacity": 0,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 0,
              "padding": 0,
              "paddingHorizontal": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(354, 70%, 54%)",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0.25,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "hsl(0, 0%, 0%)",
              "fontSize": 14,
              "fontWeight": "600",
              "letterSpacing": 0.75,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          LOGIN
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "hsl(0, 0%, 0%)",
              "fontSize": 14,
              "fontWeight": "600",
              "letterSpacing": 0.75,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          FORGOT
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
